scrapers/covid-19/govScrapers/getVietnam.js

```diff
@@ -2,33 +2,44 @@ const axios = require('axios');
 const cheerio = require('cheerio');
 const logger = require('../../../utils/logger');
 
-const columns = ['city', 'cases', 'beingTreated', 'recovered', 'deaths'];
-
 const mapRows = (_, row) => {
-	const city = { updated: Date.now() };
+	const columns = ['_', '_', '_', 'location', 'status'];
+	const city = {};
 	cheerio(row).children('td').each((index, cell) => {
 		cell = cheerio.load(cell);
 		switch (index) {
-			case 0: {
-				city[columns[index]] = cell.text().trim();
-				break;
-			}
-			default: {
-				city[columns[index]] = parseInt(cell.text().replace(/\./g, '')) || null;
+			case 3:
+			case 4:
+				city[columns[index]] = cell.text();
 				break;
-			}
 		}
 	});
 	return city;
 };
 
+const processRow = (json) => {
+	const city = [];
+	const cityList = [...new Set(json.map(tmp => tmp.location))];
```

Comment:
instead of looping over these results multiple times, do this instead: It creates a list of lists of objects, like: [[{location: a...}, {location: a...}], {location: b...}]]
```js
const ref = {};
const cityList = json.reduce(function(arr, cityObject) {
  // check index defined in `ref`
  if (ref.hasOwnProperty(cityObject.location))
  // if defined then push value
    arr[ref[cityObject.location]].push(cityObject);
  // else add index and push new array as element
  else {
    ref[cityObject.location] = arr.length;
    arr.push([cityObject]);
  }
  return arr;
}, []);
```